### PR TITLE
[feature] #286: link `api_spec` from the core repo

### LINKED ISSUE #286: Include `api_spec.md` from Iroha core repo
We can combine our `snippet_sources` with [Markdown file inclusion](https://vitepress.dev/guide/markdown#markdown-file-inclusion) feature and include API specification directly into our documentation:

```md
# API Spec

<!--@include: @/snippets/iroha2_dev_api_spec.md-->
```

### PR DESCRIPTION
**Pull request by 0x009922**
Close #286

Things to resolve:

- [x] Included markdown contains some dead links (e.g. `../../client`) which don't seamlessly integrate with our docs. Should we update the upstream source?
- [x] Document title is rendered not nicely
- [x] Ignore `snippets` directory in Vitepress' `*.md` scanning to avoid building of unnecessary pages.

### CODE REVIEW

**Comment by outoftardis** [comment]
> Included markdown contains some dead links (e.g. ../../client) which don't seamlessly integrate with our docs. Should we update the upstream source?

The relative links upstream mean that they always link to the right versions of flies in the repo. I don't think we can make them absolute.

**Comment by 0x009922** [comment]
@outoftardis, I've added a transformation step and simply re-write relative links to the absolute revision-binded links to the Iroha repo.

Is it fine?

File: src/api/index.md [hunk 1]
```diff
@@ -1,4 +1,18 @@
...
+import { useData } from 'vitepress'
 
-Refer to
-[API Specification](https://github.com/hyperledger/iroha/blob/iroha2-dev/docs/source/references/api_spec.md).
+const { page } = useData()
+</script>
+
+::: info
+
+This page contains the `api_spec.md` right from
```
Comment (outoftardis) [inline src/api/index.md:9]:
```suggestion
This page contains the copy of `api_spec.md` from
```

**Review by outoftardis — APPROVED** [review]

**Comment by outoftardis** [comment]
> @outoftardis, I've added a transformation step and simply re-write relative links to the absolute revision-binded links to the Iroha repo.
> 
> Is it fine?

Yes, that works.

**Comment by 0x009922** [comment]
I can actually remove the title in the same way... should I do this, @outoftardis?

In this case we can put it before the `info` container and the final render should look better.

File: src/api/index.md [hunk 2]
```diff
@@ -1,4 +1,18 @@
...
+import { useData } from 'vitepress'
 
-Refer to
-[API Specification](https://github.com/hyperledger/iroha/blob/iroha2-dev/docs/source/references/api_spec.md).
+const { page } = useData()
+</script>
+
+::: info
+
+This page contains the copy of `api_spec.md` from
```
Comment (WRRicht3r) [inline src/api/index.md:9]:
```suggestion
This page contains a copy of `api_spec.md` from
```

File: src/api/index.md [hunk 3]
```diff
@@ -1,4 +1,18 @@
...
+`hyperledger/iroha#iroha2-dev`. You can read the most up-to-date version of
+it on
+[GitHub](https://github.com/hyperledger/iroha/blob/iroha2-dev/docs/source/references/api_spec.md).
+
+Please note that this page was last updated at <b>{{ new Date(page.lastUpdated).toLocaleString() }}</b>.
```
Comment (WRRicht3r) [inline src/api/index.md:14]:
```suggestion
Please note this page was last updated on <b>{{ new Date(page.lastUpdated).toLocaleString() }}</b>.
```

**Review by WRRicht3r — DISMISSED** [review]

**Review by WRRicht3r — APPROVED** [review]

**Review by 6r1d — APPROVED** [review]
LGTM

File: src/api/index.md [hunk 4]
```diff
@@ -1,4 +1,18 @@
...
+</script>
+
+::: info
+
+This page contains the `api_spec.md` right from
+`hyperledger/iroha#iroha2-dev`. You can read the most up-to-date version of
+it on
+[GitHub](https://github.com/hyperledger/iroha/blob/iroha2-dev/docs/source/references/api_spec.md).
+
+Please note that this page was last updated at <b>{{ new Date(page.lastUpdated).toLocaleString() }}</b>.
```
[the remark this answers is not in the document]
Comment (0x009922) [reply src/api/index.md]: @6r1d, it supports, but AFAIK `markdown-it` will not parse `**{{ x }}**` as `<b>{{ x }}</b>`...

...or supports, you can check here: https://markdown-it.github.io/

Anyway, it doesn't matter much what syntax to use.

**Pull request merged**